src/vcl/ws.rs

```diff
@@ -86,14 +87,20 @@ impl<'a> WS<'a> {
 
     /// Same as [`WS::copy_bytes`] but adds NULL character at the end to help converts buffers into
     /// `VCL_STRING`s
-    pub fn copy_bytes_with_null<T: AsRef<[u8]>>(&mut self, src: &T) -> Result<&'a [u8], String> {
+    ///
+    /// # SAFETY
+    /// `src` must not contain NULL characters
+    pub unsafe fn copy_bytes_with_null<T: AsRef<[u8]>>(
+        &mut self,
+        src: &T,
+    ) -> Result<&'a CStr, String> {
         let buf = src.as_ref();
         let l = buf.len();
 
         let dest = self.alloc(l + 1)?;
         dest[..l].copy_from_slice(buf);
         dest[l] = b'\0';
-        Ok(dest)
+        Ok(unsafe { CStr::from_bytes_with_nul_unchecked(dest) })
```

Comment:
shouldn't we check for NULL and error out?

Reply:
@gquintard we should, but i would rather do it as part of the #46 fix.  For now this is more of a refactoring.  Baby steps :)

Reply:
yes, but this refactor both changes the API and creates another possibility of panicking.

why not just try to convert and bubble up the error?

Reply:
I added a null check right above - is that ok?

Reply:
oh, sorry, I missed that, I was expecting you to use `from_bytes_with_nul`, my bad

Reply:
i didn't want to use it after copying possibly bad data to the workspace

Reply:
fair point